### PR TITLE
 DNN-7098: Added parameter sendNotification to UserController.UpdateUser()

### PR DESCRIPTION
**Pull request by janjonas**
This is a modified pull request of https://github.com/dnnsoftware/Dnn.Platform/pull/580; see https://dnntracker.atlassian.net/browse/DNN-7098 for more detail.

### CODE REVIEW

File: DNN Platform/Library/Entities/Users/UserEventHandlers.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@ public void UserDeleted(object sender, UserEventArgs args)
 
         public void UserApproved(object sender, UserEventArgs args)
         {
+            if (!args.SendNotification) return;
             Mail.SendMail(args.User, MessageType.UserRegistrationPublic, PortalSettings.Current);
             DeleteAllNotifications(args.User.UserID);
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/UserEventHandlers.cs]: Does `DeleteAllNotifications` not need to happen if we aren't sending a new notification?

**Comment by bdukes** [comment]
outside of my one comment. this looks good to me

File: DNN Platform/Library/Entities/Users/UserEventHandlers.cs (hunk 1 above)
Comment (janjonas) [inline DNN Platform/Library/Entities/Users/UserEventHandlers.cs]: There is a `return` statement, so `DeleteAllNotifications` won't get called if `args.SendNotification` is `false`.

**Comment by janjonas** [comment]
Hi @bdukes, please see my comment below yours.

File: DNN Platform/Library/Entities/Users/UserEventHandlers.cs (hunk 1 above)
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/UserEventHandlers.cs]: Right, that's my question.  It doesn't immediately seem to me like it should be skipped (but it's definitely possible that it's correct to skip it).

Comment (janjonas) [inline DNN Platform/Library/Entities/Users/UserEventHandlers.cs]: You're right. I've added a new commit (https://github.com/janjonas/Dnn.Platform/commit/aa28f0ca6346fa239a63959cc0a295f69d8eb008) to this pull request. I also changed the name of the method to make more clear what the method does.

**Pull request merged**